core/src/main/java/org/elasticsearch/search/aggregations/pipeline/derivative/InternalDerivative.java

```diff
@@ -80,17 +92,56 @@ public Object getProperty(List<String> path) {
     @Override
     public XContentBuilder doXContentBody(XContentBuilder builder, Params params) throws IOException {
         super.doXContentBody(builder, params);
-
         if (normalizationFactor > 0) {
             boolean hasValue = !(Double.isInfinite(normalizedValue()) || Double.isNaN(normalizedValue()));
-            builder.field("normalized_value", hasValue ? normalizedValue() : null);
+            builder.field(NORMALIZED_VALUE.getPreferredName(), hasValue ? normalizedValue() : null);
             if (hasValue && format != DocValueFormat.RAW) {
-                builder.field("normalized_value_as_string", format.format(normalizedValue()));
+                builder.field(NORMALIZED_VALUE_AS_STRING.getPreferredName(), format.format(normalizedValue()));
             }
         }
         return builder;
     }
 
+    private static final ObjectParser<Map<String, Object>, Void> PARSER = new ObjectParser<>(
+            "internal_derivative", HashMap::new);
+
+    static {
+        declareCommonField(PARSER);
+        PARSER.declareField((map, value) -> map.put(CommonFields.VALUE.getPreferredName(), value),
+                (p, c) -> InternalMax.parseValue(p, Double.NEGATIVE_INFINITY), CommonFields.VALUE,
```

Comment:
InternalMax is a leftover here, although it does do the right thing. Shall we call p parser and c context?